src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt

```diff
@@ -173,16 +175,46 @@ data class SentenceCalculation(
   }
 
   val earlyReleaseSchemeEligibilityDate: LocalDate? get() {
-    if (numberOfDaysToErsed == null) {
-      return null
+    if (calculateErsed) {
+      if (sentence.identificationTrack.calculateErsedFromHalfway()) {
+        return calculateErsedFromHalfway()
+      }
+      if (sentence.identificationTrack.calculateErsedFromTwoThirds()) {
+        return calculateErsedFromTwoThirds()
+      }
     }
-    return sentence.sentencedAt
-      .plusDays(numberOfDaysToErsed.toLong())
-      .plusDays(calculatedTotalAddedDays.toLong())
-      .minusDays(calculatedTotalDeductedDays.toLong())
-      .plusDays(calculatedTotalAwardedDays.toLong())
+    return null
   }
 
+  private fun calculateErsedFromTwoThirds(): LocalDate {
+    val custodialDuration = sentence.custodialDuration()
+    val days = custodialDuration.getLengthInDays(sentence.sentencedAt)
+    return if (days >= 1097) {
+      val release = if (extendedDeterminateParoleEligibilityDate != null) extendedDeterminateParoleEligibilityDate!! else releaseDate
```

Comment:
I think you should be able to replace line 193 with the elvis operator equivalent:
`val release = extendedDeterminateParoleEligibilityDate ?:  releaseDate`

Reply:
Done